10:35:36.896 | [38;5;184mWARNING[0m | preprocess_cat - Still missing values in columns: ['age_level', 'city_development_index', 'var_1', 'user_depth']
10:35:37.106 | [38;5;184mWARNING[0m | preprocess_cat - 3019 rows still have missing 'is_click'. Dropping these rows.
10:36:51.884 | [38;5;184mWARNING[0m | preprocess_cat - Still missing values in columns: 528
C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\wandb_test3.py:14: FutureWarning: The behavior of Series.replace (and DataFrame.replace) with CategoricalDtype is deprecated. In a future version, replace will only be used for cases that preserve the categories. To change the categories, use ser.cat.rename_categories instead.
  value = value.replace('missing', np.nan)
10:37:43.347 | [38;5;184mWARNING[0m | root - Truncating wandb.Table object to 200000 rows.
[34m[1mwandb[0m: [33mWARNING[0m Serializing object of type Series that is 3754352 bytes
10:38:19.764 | [38;5;160mERROR[0m   | Flow run[35m 'grinning-goose'[0m - Encountered exception during execution: TypeError('Object of type Series is not JSON serializable')
Traceback (most recent call last):
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 746, in run_context
    yield self
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 1354, in run_flow_sync
    engine.call_flow_fn()
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 766, in call_flow_fn
    result = call_with_parameters(self.flow.fn, self.parameters)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\utilities\callables.py", line 208, in call_with_parameters
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\wandb_test3.py", line 48, in preprocess_flow
    df_clean, X_train, X_test, y_train, y_test, fold_datasets = preprocessor.preprocess(df)
                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\preprocess_cat.py", line 254, in preprocess
    self.callback({
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\wandb_test3.py", line 21, in wandb_callback
    wandb.log(metrics)
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 441, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 393, in wrapper_fn
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 383, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 1895, in log
    self._log(data=data, step=step, commit=commit)
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 1602, in _log
    self._partial_history_callback(data, step, commit)
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 1432, in _partial_history_callback
    self._backend.interface.publish_partial_history(
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\interface\interface.py", line 686, in publish_partial_history
    item.value_json = json_dumps_safer_history(v)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\util.py", line 814, in json_dumps_safer_history
    return dumps(obj, cls=WandBHistoryJSONEncoder, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\util.py", line 779, in default
    return json.JSONEncoder.default(self, obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Series is not JSON serializable
10:38:19.981 | [38;5;160mERROR[0m   | Flow run[35m 'grinning-goose'[0m - Finished in state [38;5;160mFailed[0m('Flow run encountered an exception: TypeError: Object of type Series is not JSON serializable')
Traceback (most recent call last):
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\wandb_test3.py", line 54, in <module>
    preprocess_flow("data/raw/train_dataset_full.csv", "data/processed")
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flows.py", line 1392, in __call__
    return run_flow(
           ^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 1511, in run_flow
    ret_val = run_flow_sync(**kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 1356, in run_flow_sync
    return engine.state if return_type == "state" else engine.result()
                                                       ^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 331, in result
    raise self._raised
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 746, in run_context
    yield self
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 1354, in run_flow_sync
    engine.call_flow_fn()
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\flow_engine.py", line 766, in call_flow_fn
    result = call_with_parameters(self.flow.fn, self.parameters)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\prefect\utilities\callables.py", line 208, in call_with_parameters
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\wandb_test3.py", line 48, in preprocess_flow
    df_clean, X_train, X_test, y_train, y_test, fold_datasets = preprocessor.preprocess(df)
                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\preprocess_cat.py", line 254, in preprocess
    self.callback({
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\wandb_test3.py", line 21, in wandb_callback
    wandb.log(metrics)
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 441, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 393, in wrapper_fn
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 383, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 1895, in log
    self._log(data=data, step=step, commit=commit)
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 1602, in _log
    self._partial_history_callback(data, step, commit)
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\wandb_run.py", line 1432, in _partial_history_callback
    self._backend.interface.publish_partial_history(
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\sdk\interface\interface.py", line 686, in publish_partial_history
    item.value_json = json_dumps_safer_history(v)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\util.py", line 814, in json_dumps_safer_history
    return dumps(obj, cls=WandBHistoryJSONEncoder, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\Classes\Classical_ML\YDATA-kaggle-assignment\venv\Lib\site-packages\wandb\util.py", line 779, in default
    return json.JSONEncoder.default(self, obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maorb\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Series is not JSON serializable
